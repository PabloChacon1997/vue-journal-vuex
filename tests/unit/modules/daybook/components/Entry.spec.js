import { shallowMount } from "@vue/test-utils"
import Entry from "@/modules/daybook/components/Entry"

describe("Pruebas en el Entry Component", () => {
  const mockRouter = {
    push: jest.fn(),
  }
  const entry = {
    id: "-MiT8xnGUDnzgo_9uAQs",
    date: 1630446008310,
    text: "Hola desde el mock de journal module",
  }

  const wrapper = shallowMount(Entry, {
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
    props: {
      entry,
    },
  })
  test("debe de hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test("debe de redireccionar al hacer click en el entry-container", () => {
    const classEntryContainer = wrapper.find(".entry-container")
    classEntryContainer.trigger("click")
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: {
        id: "-MiT8xnGUDnzgo_9uAQs",
      },
    })
  })

  test("pruebas en las propiedades computadas", () => {
    expect(typeof wrapper.vm.day).toBe("number")
    expect(typeof wrapper.vm.month).toBe("string")
    expect(typeof wrapper.vm.yearDay).toBe("string")
  })
})
