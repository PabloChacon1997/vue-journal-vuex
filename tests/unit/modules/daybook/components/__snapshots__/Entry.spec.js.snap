// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry Component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">31</span><span class="mx-1 fs-5">Agosto</span><span class="mx-2 fw-light">2021, Martes</span></div>
  <div class="entry-description">Hola desde el mock de journal module</div>
</div>
`;
